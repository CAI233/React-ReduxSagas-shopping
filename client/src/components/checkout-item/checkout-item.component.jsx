import React from 'react';
import { CheckoutItemStyledContainer, ImageContainer, SpanStyled, Quantity, Arrow, RemoveButton } from './checkout-item.styles';

import { connect } from 'react-redux';
import { deleteCartItem, addCartItem, lowerCartItem } from '../../redux/cart/cart.actions';

const CheckoutItem = ({ cartItem, deleteCartItem, addCartItem, lowerCartItem }) =>{
    const { name, imageUrl, price, quantity } = cartItem;
    return(
        <CheckoutItemStyledContainer>
            <ImageContainer>
                <img src={imageUrl} alt=""/>
            </ImageContainer>
            <SpanStyled >{name}</SpanStyled>
            <Quantity>
                <Arrow onClick={ ()=>lowerCartItem(cartItem) } >&#10094;</Arrow>
                    <span className="value">
                        {quantity}
                    </span>
                <Arrow onClick={ ()=>addCartItem(cartItem) } >&#10095;</Arrow>
            </Quantity>
            <SpanStyled>ï¿¥{price}</SpanStyled>
            <RemoveButton onClick={ ()=>deleteCartItem(cartItem) }>
                &#10006;
            </RemoveButton>
        </CheckoutItemStyledContainer>
    );
}

const mapDispatchToProps = dispatch => ({
    deleteCartItem: cartItem => dispatch(deleteCartItem(cartItem)),
    addCartItem: cartItem => dispatch(addCartItem(cartItem)),
    lowerCartItem: cartItem => dispatch(lowerCartItem(cartItem)),
});

export default connect(null,mapDispatchToProps)(CheckoutItem) ;