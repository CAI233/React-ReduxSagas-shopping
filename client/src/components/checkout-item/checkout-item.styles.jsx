import styled from 'styled-components';

export const CheckoutItemStyledContainer = styled.div`
  width: 100%;
  display: flex;
  min-height: 100px;
  border-bottom: 1px solid darkgrey;
  padding: 15px 0;
  font-size: 20px;
  align-items: center;
`;

CheckoutItemStyledContainer.displayName = "CheckoutItemStyledContainer";

export const ImageContainer = styled.div`
    width: 23%;
    padding-right: 15px;
    overflow: hidden;

    img {
      width: 100%;
      height: 100%;
    }
`;

ImageContainer.displayName = "ImageContainer";

export const SpanStyled = styled.span`
    width:23%;
`;

SpanStyled.displayName = "SpanStyled";

// styled( 标签 )也可以继承其它styled标签的属性( 完成笔记 )
export const Quantity = styled(SpanStyled)`
    padding-left: 20px;
    display: flex;
`;

Quantity.displayName = "Quantity";

export const Arrow = styled.div`
    margin:0 0.3rem;
    cursor: pointer;

    // 取消标签双击选中
    -moz-user-select:none;
    -webkit-user-select:none;
    -ms-user-select:none;
    -khtml-user-select:none;
    user-select:none;
`;

Arrow.displayName = "Arrow";

export const RemoveButton = styled.span`
    padding-left: 12px;
    cursor: pointer;
`;

RemoveButton.displayName = "RemoveButton";