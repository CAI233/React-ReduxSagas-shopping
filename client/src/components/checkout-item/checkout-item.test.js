import { shallow } from 'enzyme';
import React from 'react';
import { CheckoutItem } from './checkout-item.component';

describe("测试结算页面单个商品显示",()=>{
    let wrapper,mockProps;
    const mockItem = {
        imageUrl: '1.jpg',
        name: 'test',
        price: 10,
        quantity: 1,
    };
    beforeEach( ()=>{
        mockProps = {
            cartItem: mockItem ,
            deleteCartItem: jest.fn(), 
            addCartItem: jest.fn(), 
            lowerCartItem: jest.fn() 
        }
        wrapper = shallow(<CheckoutItem { ...mockProps } />);
    } );

    it("CheckoutItem Snapshot",()=>{
        expect(wrapper).toMatchSnapshot();
    });

    it("模拟单击删除测试",()=>{
        wrapper.find("RemoveButton").simulate('click');
        expect(mockProps.deleteCartItem).toHaveBeenCalled();
    });

    it("模拟单击添加产品数量测试",()=>{
        wrapper.find("Quantity").childAt(0).simulate('click'); // childAt( number ): 通过子标签排位数，从上往下，从0~n，抓取子标签( 等待笔记 )
        expect(mockProps.lowerCartItem).toHaveBeenCalled();
    });

    it("模拟单击减少产品数量测试",()=>{
        wrapper.find("Quantity").childAt(2).simulate('click'); 
        expect(mockProps.addCartItem).toHaveBeenCalled();
    });

});