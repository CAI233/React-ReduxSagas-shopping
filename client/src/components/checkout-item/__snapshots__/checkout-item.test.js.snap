// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试结算页面单个商品显示 CheckoutItem Snapshot 1`] = `ShallowWrapper {}`;
